crates/driver/src/solver/mod.rs

```diff
@@ -0,0 +1,89 @@
+use {
+    crate::{
+        logic::{
+            competition::{
+                auction::{self, Auction},
+                solution::Solution,
+            },
+            eth,
+        },
+        util,
+    },
+    thiserror::Error,
+};
+
+mod dto;
+
+/// Solvers are controlled by the driver. Their job is to search for solutions
+/// to auctions. They do this in various ways, often by analyzing different AMMs
+/// on the Ethereum blockchain.
+pub struct Solver {
+    url: reqwest::Url,
+    client: reqwest::Client,
+    /// Used for building the instance name to send to the solver.
+    network_name: eth::NetworkName,
+    /// Used for building the instance name to send to the solver.
+    chain_id: eth::ChainId,
+}
+
+const MAX_NR_EXEC_ORDERS: &str = "100";
+const SOLVER_RESPONSE_MAX_BYTES: usize = 10_000_000;
+
+impl Solver {
+    pub fn new(url: reqwest::Url, network_name: eth::NetworkName, chain_id: eth::ChainId) -> Self {
+        let mut headers = reqwest::header::HeaderMap::new();
+        headers.insert(
+            reqwest::header::CONTENT_TYPE,
+            "application/json".parse().unwrap(),
+        );
+        headers.insert(reqwest::header::ACCEPT, "application/json".parse().unwrap());
+        // TODO(#907) Also add an auth header
+        Self {
+            url,
+            client: reqwest::ClientBuilder::new()
+                .default_headers(headers)
+                .build()
+                .unwrap(),
+            network_name,
+            chain_id,
+        }
+    }
+
+    pub async fn solve(&self, auction: Auction) -> Result<Solution, Error> {
+        // TODO Ask about all the `config` stuff in DefaultHttpSolverApi, what is every
+        // field for exactly?
+        let mut url = self.url.join("solve").unwrap();
+        let time_limit = auction.deadline.solver_time_limit()?;
+        url.query_pairs_mut()
+            .append_pair("auction_id", &auction.id.0.to_string())
+            .append_pair("instance_name", &self.instance_name(auction.id))
```

Comment:
For example, this is literally telling the solver what to name the instance file. I think you could get it do crazy things by specifying a path (`../../etc/passwd` 😛).